lib/client.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+const Request = require('./request');
+const SwaggerClient = require('./swagger-client');
+
+class Client extends SwaggerClient {
+  constructor(options) {
+    const http = new Request(options.config);
+    return new Promise((resolve, reject) => {
+      http.request('GET', { path: '/swagger.json' }, (err, res) => {
+        if (err) return reject(err);
+        if (res.statusCode !== 200) {
+          return reject(new Error(`Failed to get swagger.json: ${ res.statusCode }`));
+        }
+        resolve(super({ http, spec: res.body }));
```

Comment:
This is going to cause problems for Node < 6.  Should we drop support for Node 4 and Node 5?

Reply:
Node 4 EOLs at the end of April : https://github.com/nodejs/Release , seems fine to not go out of our way at this point? It is slightly sad that we won't hit the EOL date, but don't see much sense in going out of our way for this.